src/Compilers/Core/Portable/CodeGen/PermissionSetAttribute.cs

```diff
@@ -80,7 +81,7 @@ public ImmutableArray<Cci.IMetadataNamedArgument> GetNamedArguments(EmitContext
             Debug.Assert(!String.IsNullOrEmpty(fileName));
 
             // PermissionSetAttribute type must have a writable public string type property member 'Hex'
-            Debug.Assert(((INamedTypeSymbol)_sourceAttribute.GetType(context)).GetMembers(HexPropertyName).Any(
+            Debug.Assert(((INamedTypeSymbol)((ISymbolInternal)_sourceAttribute.GetType(context)).GetISymbol()).GetMembers(HexPropertyName).Any(
```

Comment:
There's a lot of casting going on on this line. Consider breaking out some of this for readability.

Reply:
> There's a lot of casting going on on this line. Consider breaking out some of this for readability.

Do you have some breakout in mind? It might be hard to guess what you would consider more readable.


---
In reply to: [338835825](https://github.com/dotnet/roslyn/pull/39498#discussion_r338835825) [](ancestors = 338835825)

Reply:
Perhaps pulling the `((ISymbolInternal)_sourceAttribute.GetType(context)).GetISymbol()` section out? My main issue is that with this number of parens, it's hard to tell what's surrounding what.

---
In reply to: [338836192](https://github.com/dotnet/roslyn/pull/39498#discussion_r338836192) [](ancestors = 338836192,338835825)